neuralpp/symbolic/sympy_interpreter.py

```diff
@@ -81,18 +77,9 @@ def simplify(self, expression: Expression, context: Expression = BasicConstant(T
         if not isinstance(expression, SymPyExpression):
             expression = SymPyExpression.convert(expression)
 
-        result = SymPyInterpreter._simplify_expression(expression.sympy_object, context)
-        # if "x+y" is simplified to "x+2", how do we find out the type of that "+"? We need to update the type_dict
-        # by also doing the replacement. If we have "x+(y+y)", the type_dict would be
-        #   {x:int, y:int, y+y:int->int->int, x+(y+y):int->int->int}
-        # by a context of {y:2}, we'd have
-        #   {x:int, 2:int, 4:int->int->int, x+4:int->int->int} (entry 2 and 4 will be GCed)
-        type_dict = SymPyInterpreter._updated_type_dict_by_context(expression.type_dict, context)
```

Comment:
Don't need to update_type_dict_by_context, because `type_dict` only contains "leaf atoms" (including uninterpreted functions), whose type should not change during simplification (but they could have been deleted, so call `purge_type_dict` next.